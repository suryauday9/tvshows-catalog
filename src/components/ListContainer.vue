<template>
  <div>
    <v-sheet class="mx-auto white darken-4">
      <div class="headline font-weight-bold blue-grey--text mx-12 mt-12">
        {{ genre }}
      </div>
      <v-slide-group class="pa-4" show-arrows="desktop" light>
        <v-slide-item v-for="show in showsByGenre(shows, genre)" :key="show.id">
          <show-preview :show="show" />
        </v-slide-item>
      </v-slide-group>
    </v-sheet>
  </div>
</template>

<script>
import ShowPreview from "../components/ShowPreview.vue";
export default {
  name: "ListContainer",
  components: {
    ShowPreview
  },
  props: {
    genre: String
  },
  methods: {
    showsByGenre: function (shows, genre) {
      return shows
        .filter((show) => show.genres && show.genres.some((g) => g === genre))
        .slice(0, 19)
    }
  },
  computed: {
    shows() {
        return this.$store.state.shows;
    }
  }
};
</script>
<style scoped>
.headline {
  text-align: left;
  text-decoration: underline;
}
</style>