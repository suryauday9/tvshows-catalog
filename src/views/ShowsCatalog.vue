<template>
    <div>
        <div class="searchshow-container">
            <search-shows></search-shows>
        </div>
        <div class="genre-shows-container" v-if="genres.length">
            <div v-for="(list, index) in genres" :key="index">
                <list-container :genre="list" />
            </div>
        </div>
        <div v-else>
            No shows found / No shows found with this search term.
        </div>
    </div>
</template>

<script>
import SearchShows from '../components/SearchShows.vue';
import ListContainer from "../components/ListContainer.vue";

export default {
    name: 'ShowsCatalog',
    components: {
        SearchShows,
        ListContainer
    },
    methods: {
        initialize() {
            this.$store.dispatch('getShowslist');
        }
    },
    created() {
        this.initialize();
    },
    computed: {
        genres() {
            return this.$store.state.genres;
        }
    }
};
</script>

<style scoped></style>
