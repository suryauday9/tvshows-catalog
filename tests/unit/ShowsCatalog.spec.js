import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import { initializeApp } from './initializeApp';
import { showsData, genres } from './response';
import ShowsCatalog from '../../src/views/ShowsCatalog.vue';
import ListContainer from '../../src/components/ListContainer.vue';

const vuetify = new Vuetify();
const localVue = initializeApp(createLocalVue());

describe('ShowsCatalog component', () => {  
    let wrapper;
    
    const store = new Vuex.Store({
        state: {
            shows: showsData,
            genres: genres,
            showDetails: [],
            showThumbnails: [],
            showSearchbar: false
        },
        actions: {
            getShowslist: jest.fn()
        }
    });

    const mountComponent = (options) => {
        return shallowMount(ShowsCatalog, {
            localVue,
            store,
            vuetify,
            ...options,
        })
    }

    beforeEach(() => {
        wrapper = mountComponent();
    })

    afterEach(() => {
        wrapper.destroy();
    })

    it('renders component correctly', () => {
        // see if component renders
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should be a Vue instance', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    it('should trigger initialize method', () => {
        const actn = wrapper.vm.initialize();
        expect(actn).toBeUndefined();
        expect(wrapper.vm.$nextTick()).toStrictEqual(store.dispatch('getShowslist'));
    });

    it('rendered component data should match with store', () => {
        // see if data loads
        expect(wrapper.vm.genres).toBe(store.state.genres);
    });

    it('it should load ListContainer component', () => {
        expect(ListContainer).toBeTruthy();
    });

});