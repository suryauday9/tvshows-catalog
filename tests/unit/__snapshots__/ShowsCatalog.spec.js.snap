// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowsCatalog component renders component correctly 1`] = `
<div>
  <div
    class="searchshow-container"
  >
    <search-shows-stub />
  </div>
   
  <div
    class="genre-shows-container"
  >
    <div>
      <list-container-stub
        genre="Action"
      />
    </div>
    <div>
      <list-container-stub
        genre="Crime"
      />
    </div>
    <div>
      <list-container-stub
        genre="Drama"
      />
    </div>
    <div>
      <list-container-stub
        genre="Horror"
      />
    </div>
    <div>
      <list-container-stub
        genre="Science-Fiction"
      />
    </div>
    <div>
      <list-container-stub
        genre="Thriller"
      />
    </div>
  </div>
</div>
`;
